Add support for live documentation editing

lewisdenny opened the pull request:
This patch adds the docs-live make target. This make target uses
`sphinx-autobuild` to rebuild and reload docuemtnation content in the
browser.

As a pull request owner and reviewers, I checked that:
- [x] Appropriate testing is done and actually running

jpodivin commented on Makefile:
```diff
@@ -223,6 +223,10 @@ enable-git-hooks:
 docs: ## Create documentation under docs/build/html
 	./docs/source/build-docs.sh
 
+.PHONY: docs-live
```
nit: the convention in this makefile is to use underscores `_` rather than dashes `-`.

jpodivin reviewed (changes requested):
The plugins, roles and utils are not getting build for live version. 
```
/home/jpodivin/repos/ci-framework/docs/source/index.rst:95: WARNING: toctree glob pattern 'plugins/*' didn't match any documents
/home/jpodivin/repos/ci-framework/docs/source/index.rst:102: WARNING: toctree glob pattern 'module_utils/*' didn't match any documents
/home/jpodivin/repos/ci-framework/docs/source/index.rst:109: WARNING: toctree glob pattern 'roles/*' didn't match any documents

You can work around it by building the standard version first, and using the `sphinx-autobuild` just to serve it. But I'm not sure if the changes are going to propagate that way.

lewisdenny commented:
> The plugins, roles and utils are not getting build for live version.
> 
> ```
> /home/jpodivin/repos/ci-framework/docs/source/index.rst:95: WARNING: toctree glob pattern 'plugins/*' didn't match any documents
> /home/jpodivin/repos/ci-framework/docs/source/index.rst:102: WARNING: toctree glob pattern 'module_utils/*' didn't match any documents
> /home/jpodivin/repos/ci-framework/docs/source/index.rst:109: WARNING: toctree glob pattern 'roles/*' didn't match any documents
> 
> You can work around it by building the standard version first, and using the `sphinx-autobuild` just to serve it. But I'm not sure if the changes are going to propagate that way. 
> ```

Thanks @jpodivin , I have added a new docs make target to build the auto generated content. I then pass that to `sphinx-autobuild as a `--pre-build` command. I also added the source plugins directory as one to watch for changes and excluded the generated content from the watch as it causes a loop. 

Please test and let me know if that helps the issues you discovered!

jpodivin commented:
> > The plugins, roles and utils are not getting build for live version.
> > ```
> > /home/jpodivin/repos/ci-framework/docs/source/index.rst:95: WARNING: toctree glob pattern 'plugins/*' didn't match any documents
> > /home/jpodivin/repos/ci-framework/docs/source/index.rst:102: WARNING: toctree glob pattern 'module_utils/*' didn't match any documents
> > /home/jpodivin/repos/ci-framework/docs/source/index.rst:109: WARNING: toctree glob pattern 'roles/*' didn't match any documents
> > 
> > You can work around it by building the standard version first, and using the `sphinx-autobuild` just to serve it. But I'm not sure if the changes are going to propagate that way. 
> > ```
> 
> Thanks @jpodivin , I have added a new docs make target to build the auto generated content. I then pass that to `sphinx-autobuild` as a `--pre-build` command. I also added the source plugins directory as one to watch for changes and excluded the generated content from the watch as it causes a loop.
> 
> Please test and let me know if that helps the issues you discovered!

Thanks. I've verified that the documentation includes all parts of the CIFMW and that it's possible to edit it live. Great work.

cjeanner reviewed (approved):
/approve
